AssertionError: Accessing object which has been invalidated or deleted

View details in Rollbar: [https://app.rollbar.com/a/sa.jansen1/fix/item/hymnbook/320](https://app.rollbar.com/a/sa.jansen1/fix/item/hymnbook/320)

```

AssertionError: Accessing object which has been invalidated or deleted
  File "http://reactnativehost/address at index.android.bundle", line 1, in callFunctionReturnFlushedQueue
  File "http://reactnativehost/address at index.android.bundle", line 1, in __guard
  File "http://reactnativehost/address at index.android.bundle", line 1, in anonymous
  File "http://reactnativehost/address at index.android.bundle", line 1, in __callFunction
  File "http://reactnativehost/(native)", line unknown, in apply
  File "http://reactnativehost/address at index.android.bundle", line 1, in receiveEvent
  File "http://reactnativehost/address at index.android.bundle", line 1, in _receiveRootNodeIDEvent
  File "http://reactnativehost/address at index.android.bundle", line 1, in batchedUpdates
  File "http://reactnativehost/address at index.android.bundle", line 1, in batchedUpdatesImpl
  File "http://reactnativehost/address at index.android.bundle", line 1, in flushSyncCallbacks
  File "http://reactnativehost/address at index.android.bundle", line 1, in performSyncWorkOnRoot
  File "http://reactnativehost/address at index.android.bundle", line 1, in renderRootSync
  File "http://reactnativehost/address at index.android.bundle", line 1, in workLoopSync
  File "http://reactnativehost/address at index.android.bundle", line 1, in performUnitOfWork
  File "http://reactnativehost/address at index.android.bundle", line 1, in beginWork$1
  File "http://reactnativehost/address at index.android.bundle", line 1, in updateFunctionComponent
  File "http://reactnativehost/address at index.android.bundle", line 1, in renderWithHooks
  File "http://reactnativehost/address at index.android.bundle", line 1, in SingleDocument
  File "http://reactnativehost/address at index.android.bundle", line 1, in get
  File "http://reactnativehost/address at index.android.bundle", line 1, in get
  File "http://reactnativehost/address at index.android.bundle", line 1, in anonymous
  File "http://reactnativehost/address at index.android.bundle", line 1, in assert
  File "http://reactnativehost/address at index.android.bundle", line 1, in AssertionError
  File "http://reactnativehost/(native)", line unknown, in call
  File "http://reactnativehost/address at index.android.bundle", line 1, in _createSuperInternal
  File "http://reactnativehost/(native)", line unknown, in construct
  File "http://reactnativehost/address at index.android.bundle", line 1, in Wrapper
  File "http://reactnativehost/address at index.android.bundle", line 1, in _construct
  File "http://reactnativehost/(native)", line unknown, in apply
  File "http://reactnativehost/(native)", line unknown, in construct
```

--- a/source/gui/screens/documents/document/SingleDocument.tsx
+++ b/source/gui/screens/documents/document/SingleDocument.tsx
@@ -50,7 +50,7 @@ const SingleDocument: React.FC<NativeStackScreenProps<ParamList, typeof Document
   const fadeInFallbackTimeout = useRef<NodeJS.Timeout | undefined>();
   const htmlViewLastLoadedForDocumentId = useRef<number | undefined>();
 
-  const [document, setDocument] = useState<(Document & Realm.Object<Document>) | undefined>(undefined);
+  const [document, setDocument] = useState<Document | undefined>(undefined);
   const [scrollOffset, setScrollOffset] = useState(0);
   const [bottomOffset, setBottomOffset] = useState(999);
   const [onPressed, setOnPressed] = useState(false);
@@ -98,7 +98,7 @@ const SingleDocument: React.FC<NativeStackScreenProps<ParamList, typeof Document
   }, [document?.id]);
 
   useLayoutEffect(() => {
-    if (document === undefined) {
+    if (document == undefined) {
       navigation.setOptions({
         title: ""
       });
@@ -127,7 +127,8 @@ const SingleDocument: React.FC<NativeStackScreenProps<ParamList, typeof Document
       // Document not found
     }
 
-    setDocument(newDocument);
+    setDocument(newDocument ? Document.clone(newDocument, { includeParent: true }) : undefined);
+    setDocument(newDocument ? Document.clone(newDocument, { includeParent: true }) : undefined);
   };
 
   const animate = (callback?: () => void) => {

